osu.Game/Overlays/Settings/Sections/Maintenance/StableDirectoryLocationDialog.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Threading.Tasks;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Screens;
+using osu.Game.Overlays.Dialog;
+
+namespace osu.Game.Overlays.Settings.Sections.Maintenance
+{
+    public class StableDirectoryLocationDialog : PopupDialog
+    {
+        [Resolved]
+        private OsuGame game { get; set; }
+
+        public StableDirectoryLocationDialog(TaskCompletionSource<string> taskCompletionSource)
+        {
+            HeaderText = "Failed to automatically locate a stable installation.";
+            BodyText = "osu! failed to automatically locate a stable installation. Maybe you can tell osu! where it is located?";
+            Icon = FontAwesome.Solid.QuestionCircle;
+
+            Buttons = new PopupDialogButton[]
+            {
+                new PopupDialogOkButton
+                {
+                    Text = "Sure! I know where it is located!",
+                    Action = () => Schedule(() => game.PerformFromScreen(screen => screen.Push(new StableDirectorySelectScreen(taskCompletionSource))))
```

Comment:
Does this need to transition completely to another screen just to select the path for osu!stable install? sounds like it should just pop up an overlay in the current screen, probably inside the dialog or somewhere near.

Reply:
I don't personally think it matters whether it's a separate screen or an overlay. And I'd much rather have one screen used in two places than a screen used in one place and an overlay in another.

Reply:
The only advantage that having an overlay would have over a screen in this situation would be the fact that you won't get sent back to the main menu once you're done with the selection, imo but I'm not sure we can consider this as a problem in the first place?

Reply:
It seems it isn't easily possible to add a directory selector to the popup dialog (fixed to one width by the overlay itself and designed to not allow custom content). 

I guess this is fine for now then, but later on, this probably needs some type of `StoragePathSelectorOverlay` that can be resolved to request a file/directory selection.